packages/runtime/test/integration/deploy-asa.ts

```diff
@@ -0,0 +1,36 @@
+import { assert } from "chai";
+
+import { AccountStore, Runtime } from "../../src/index";
+import { useFixture } from "../helpers/integration";
+
+const minBalance = 1e6;
+
+describe("Deploy ASA with mutiple opt-in accounts", function () {
+  useFixture("deploy-asa");
+  let john: AccountStore;
+  let bob: AccountStore;
+  let runtime: Runtime;
+  let alice: AccountStore;
+
+  this.beforeAll(async function () {
+    john = new AccountStore(minBalance, 'john');
+    bob = new AccountStore(minBalance, 'bob');
+    alice = new AccountStore(minBalance, 'alice');
+    runtime = new Runtime([john, bob, alice]);
+  });
+
+  // helper function
+  function syncAccounts (): void {
+    john = runtime.getAccount(john.address);
+    bob = runtime.getAccount(bob.address);
+    alice = runtime.getAccount(alice.address);
+  }
+
+  it("Should opt-in to multiple accounts mentioned in asa.yaml", () => {
+    const asset = runtime.addAsset('asa', { creator: { ...john.account, name: "john" } });
```

Comment:
if `john.account` includes name(=john) as well, then can we directly use `{ creator: john.account }` here?

Reply:
no, because name is optional in john.account but while adding a asset it is required

Reply:
hmm, so this is a types issue. Can we use `{ creator: john.account as Account }` ? Because we're already initializing account by name, so passing name again seems extra here. Anyway, not a big deal.

Reply:
the problem here is that name is optional, so we will have to use a type where name is not optional.

Reply:
[https://github.com/scale-it/algo-builder/blob/master/packages/runtime/src/types.ts#L435](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/src/types.ts#L435)